rust/src/lib/nm/conn_matcher.rs

```diff
@@ -232,6 +233,53 @@ impl NmConnectionMatcher {
         nm_conns.pop()
     }
 
+    /// Find the best saved NmConnection in the order of:
+    /// * Currently activated
+    /// * Biggest `connection.autoconnect-priority`
+    /// * Biggest `connection.timestamp`
+    /// * Biggest `connection.uuid`
+    pub(crate) fn get_prefered_saved<'s>(
+        &'s self,
+        base_iface: &BaseInterface,
+    ) -> Option<&'s NmConnection> {
+        let nm_iface_type = NmIfaceType::from(&base_iface.iface_type);
+
+        if let Some(nm_conn) = self
+            .acs_by_name_and_type
+            .get(&(base_iface.name.to_string(), nm_iface_type.clone()))
+            .and_then(|nm_ac| self.saved_by_uuid.get(nm_ac.uuid.as_str()))
+            .map(Rc::as_ref)
+        {
+            return Some(nm_conn);
+        }
```

Comment:
This block seems unnecessary, and maybe wrong. If user requests match by MAC, this block might get a match by ifname first. I'm not really sure how that can actually happen, I guess that a different NIC has to be called `port1`, which is quite confusing, at the very least, or maybe directly wrong user's desired state. But anyway, I think this block should be removed.

Reply:
Please add a comment explaining this

Reply:
Command added:

```
        // Prefer the current activated NmConnection by searching
        // NmActiveConnection for specified interface, then use the UUID of
        // NmActiveConnection to search out the NmConnection.

```